test/fixtures/s3-fixture/src/main/java/fixture/s3/junit/S3HttpFixtureRule.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package fixture.s3.junit;
+
+import fixture.s3.S3HttpFixture;
+import fixture.s3.S3HttpFixtureWithEC2;
+import fixture.s3.S3HttpFixtureWithECS;
+import fixture.s3.S3HttpFixtureWithSTS;
+import fixture.s3.S3HttpFixtureWithSessionToken;
+
+import org.junit.rules.ExternalResource;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class S3HttpFixtureRule extends ExternalResource {
+    private final boolean enabled;
+    private final List<Type> enabledFixtures;
+
+    private Map<Type, S3HttpFixture> configuredFixtures = new HashMap<>();
+
+    private void configureFixture(Type fixtureType) throws Exception {
+        InetSocketAddress inetSocketAddress = resolveLoopBackAddress();
```

Comment:
I'm missing something obvious here. How is it possible for us to start multiple fixtures using the same socket address? Won't there be port conflicts?

Reply:
Oh I see now, we create a new socket address for each fixture, got it.